proto/spire/plugin/agent/trustbundlesource/v1/trustbundlesource.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+package spire.plugin.agent.trustbundlesource.v1;
+option go_package = "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/agent/trustbundlesource/v1;svitrustbundlesourcev1";
+
+import "google/protobuf/timestamp.proto";
+import "spire/plugin/types/bundle.proto";
+    
+service TrustBundleSource {
+    // Get an up to date Trust Bundle
+    rpc GetTrustBundle(GetTrustBundleRequest) returns (GetTrustBundleResponse);
+}
+
+message GetTrustBundleRequest {
+    // The trust domain for which to retrieve the trust bundle
+    string trust_domain = 1;
+
+    // Hint for which type of action this trust bundle will be used for
+    Use use_hint = 2;
+
+    // Optional server url that is being used with the retrieved bundle
+    string server_url = 3;
+
+    // How many connection attempts to the server have been made so far
+    int32 connection_attempts = 4;
+
+    // When was (re)bootstrapping started
+    google.protobuf.Timestamp starttime = 5;
```

Comment:
How is the plugin using this information? Some of these seem like plugin specific configuration, which can be done through the plugin data config, for example the server url which wouldn't be used for a file based bundle source.

Reply:
Its metadata that plugin might want to use in its calculation for returning trust bundles / insecure mode, but it doesn't have to.

For example, a plugin that just reads from a file can ignore all of this.

A policy plugin may want to look at these things to determine, pull from source A if its attempted less then 10 connections, then use source B. Or, if now() > starttime + 10m, use source C